setup.py

```diff
@@ -1,10 +1,16 @@
 import os
 import pathlib
 import sys
+from sys import platform
 
 import setuptools
 from Cython.Build import cythonize
 
+IS_LINUX = "linux" in platform
+
+if not IS_LINUX:
+    raise RuntimeError(f"memray does not support this platform ({platform})")
```

Comment:
If the `setup.py` fails with a `RuntimeError`, won't `pip install pystack` just fall back to installing an older version that didn't have this check?

Reply:
Hummmm, IIRC not anymore, but I am not sure. @pradyunsg do you know if this has changed or I am making this up?

Reply:
Not since https://pip.pypa.io/en/stable/news/#v22-0.

> Stop backtracking on build failures, by instead surfacing them to the user and aborting immediately. This behaviour provides more immediate feedback when a package cannot be built due to missing build dependencies or platform incompatibility. ([#10655](https://github.com/pypa/pip/issues/10655))

Reply:
Thanks @pradyunsg!